File: solved.ac/code/5698.js
const fs = require("fs")
let input = fs.readFileSync("./dev/stdin").toString().trim().split("\n")
// console.log(input)
solution(input)
function solution(testcaseArray) {
  const sentenceArr = testcaseArray
    .filter((p) => p !== "*")
    .map((p) => p.toLowerCase())

  sentenceArr.forEach((sentence) => {
    const firstCharArr = sentence.split(" ").map((w) => w.substr(0, 1))
    const uniqueArr = [...new Set(firstCharArr)]
    if (uniqueArr.length === 1) {
      console.log("Y")
    } else {
      console.log("N")
    }
  })
}

//         const uniqArr = [...new Set(firstCharArr)]
//         if (uniqArr.length === 1) {
//             console.log('Y')
//         } else {
//             console.log('N')
//         }
//     })
// }

File: solved.ac/code/5698.py
import sys
input = sys.stdin.readline


while True:

  s = input().rstrip().split()
  if s[0] == '*' : break
  temp =  s[0][0]
  flag = True
  for i in range(len(s)):
    if temp.upper() == s[i][0].upper():
      continue
    else:
      flag = False

  if flag:
    print("Y")
  else:
    print("N")